tests/conftest.py

```
@@ -0,0 +1,355 @@
+                )
+            )
+
+        # Skip integration tests if no API key
+        if item.get_closest_marker("integration") and not config._fmp_api_key:  # type: ignore
+            item.add_marker(pytest.mark.skip(reason="FMP_TEST_API_KEY not set"))
+
+        # Skip tests requiring API keys
+        if item.get_closest_marker("requires_api_key") and not config._fmp_api_key:  # type: ignore
+            item.add_marker(pytest.mark.skip(reason="FMP_TEST_API_KEY not set"))
+
+        if item.get_closest_marker("requires_openai_key") and not config._openai_api_key:  # type: ignore
```

Ruff (0.11.9)

Line too long (89 > 88)

(E501)